### ISSUE TITLE
`indent`: invalid multiline comments indentation

### ISSUE THREAD

**Issue by yamavol** [eslint-stylistic/eslint-stylistic#778]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

As [already reported few years ago](https://github.com/eslint/eslint/issues/13441), the indent plugin does not correctly fix the comment block indent.

```js
// before fix
    /** 
     *  comment
     */
    function b() {} 
```
```js
// after fix
/**
     *  comment
     */
function b() {} 
```
It can be reproduced with the default, minimal plugin configuration.

<details>
<summary>eslint.config.js</summary>

```
import stylistic from "@stylistic/eslint-plugin";

export default [
  {
    plugins: {
      "@stylistic": stylistic,
    },
    files: ["**/*.js"],
    rules: {
      "@stylistic/indent": ["error", 2],
    },
  },
];
```
- eslint: 9.2.0
- eslint-stylistic: 4.2.0

</details>

**Reason**

When eslint tokenizes the file, it seems there are two kinds of comment tokens.
- "line" token which starts from //
- "block" token enclosed by /* and */, keeping comments (possibly multilined) as its value

The indent rule only fixes the indent where the token starts, which is invalid in mutlilined comment.
https://github.com/eslint-stylistic/eslint-stylistic/blob/v4.2.0/packages/eslint-plugin/rules/indent/indent._js_.ts#L751-L756

This suggests indent rule should be aware of multiline tokens.
```js
// something like this?
* fix(fixer) {
  if (token.loc.start.line !== token.loc.end.line) {
    for (const line of lines) {
      yield fixer.replaceTextRange(range, newText)
    }
  }
  else {
    const range = [token.range[0] - token.loc.start.column, token.range[0]] as [number, number]
    const newText = neededIndent

    yield fixer.replaceTextRange(range, newText)
  }
},
```

I noticed the fix was not simple, because the indent rule must be aware of multiline-comment-style and the offset for each lines inside the token must be calculated.

## mutliline-comment-style

As a workaround or intended, user may set [this rule](https://eslint.style/rules/js/multiline-comment-style) to enforce correct indentation for the comment blocks. Unfortunately setting this rule enforces user to use single and consistent comment style, so some user (well, me) cannot accept this rule. Adding a new option to allow multiple comment styles might be another solution.  
example: https://github.com/eslint/eslint/issues/11975  

I might try create a PR, but then I want to discuss how it should be fixed.

### Reproduction

https://stackblitz.com/edit/github-5wn7jsnn?file=src%2Findex.ts

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
